### PR TITLE
Add doctest to our CI workflow

### PR DESCRIPTION
**Pull request by mpenkov**
This will automatically run `doctest` on README.rst for each build.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -41,6 +41,7 @@ install:
 script:
   - flake8 --max-line-length=110
   - python check_keys.py
+  - python -m doctest README.rst
```
Comment (menshikh-iv) [inline .travis.yml]:
```suggestion
  - python -m doctest README.rst -v
```

I propose to add verbosity here to 
- better understanding, what's going on on this stage
- make sure than a fresh "piece of code" was interpreted as docstring (to avoid the situation that happened yesterday).

File: README.rst [hunk 2]
```diff
@@ -133,15 +133,22 @@ More examples
 
 .. code-block:: python
 
-    >>> import boto3
+    >>> import os, sys, boto3
```
Comment (menshikh-iv) [inline README.rst]: also, I thought that this is a good idea to get rid `repr` function that used for strings (we don't really need them), see my "pathlib-patching" example

File: README.rst [hunk 3]
```diff
@@ -40,18 +40,12 @@ How?
   >>> from smart_open import open
   >>>
   >>> # stream lines from an S3 object
-  >>> for line in open('s3://commoncrawl/robots.txt'):
-  ...    print(repr(line))
-  ...    break
+  >>> open('s3://commoncrawl/robots.txt').readline()
```
Comment (piskvorky) [inline README.rst]: Strong -1: show line iteration in the examples, as prominently as possible.

Why this change?

File: README.rst [hunk 4]
```diff
@@ -133,15 +123,22 @@ More examples
 
 .. code-block:: python
 
-    >>> import boto3
+    >>> import os, sys, boto3
+    >>>
+    >>> def get_session():
+    ...     # This is here so we can doctest this example under TravisCI
```
Comment (piskvorky) [inline README.rst]: Mild -1: the purpose of README is clear communication, not doctests.

If the two goals clash, clarity of message takes precedence. Doctests less important.

File: README.rst [hunk 5]
```diff
@@ -313,7 +310,7 @@ The ``version_id`` transport parameter enables you to get the desired version of
 
   >>> # If you don't specify a version, smart_open will read the most recent one
   >>> with open('s3://%s/%s' % (bucket, key)) as fin:
-  ...     print(repr(fin.read()))
+  ...     fin.read()
```
Comment (piskvorky) [inline README.rst]: Why this change? Intent less clear.

File: README.rst (hunk 4 above)
Comment (menshikh-iv) [reply README.rst]: btw if we have broken example in the docstring, that's bad for users, for this reason, I propose to make something similar as in gensim we done: some "internal util functions" inside package that "make dirty work" and just re-use them in docstring (similar stuff in gensim are `datapath`, small corpora & dictionaries, etc).

File: README.rst (hunk 3 above)
Comment (mpenkov) [reply README.rst]: Two reasons:

1. Brevity
2. A for-loop that breaks after the first iteration is redundant

File: README.rst (hunk 5 above)
Comment (mpenkov) [reply README.rst]: Again, brevity. `print(repr(...))` is the default behavior of a REPL session.

File: README.rst (hunk 3 above)
Comment (mpenkov) [reply README.rst]: We show line iteration in separate examples (e.g. the example below), so there's little value in showing it again here, in my opinion.

Comment (piskvorky) [reply README.rst]: No, iteration should be the top-most (most visible, most prominent) example.

Also showing it "below" is fine too, to drive the point home.

File: README.rst (hunk 5 above)
Comment (piskvorky) [reply README.rst]: Our README goal is clarity, not brevity.

There is no assumption the code goes into REPL. Keep the `print` please.

File: README.rst (hunk 3 above)
Comment (mpenkov) [reply README.rst]: OK, reverted.

File: README.rst (hunk 5 above)
Comment (mpenkov) [reply README.rst]: OK, reverted.
